probes/pinsDependencies/impl_test.go

```diff
@@ -0,0 +1,669 @@
+					Dependencies: []checker.Dependency{
+						{
+							Location: &checker.File{
+								Snippet: "actions/checkout@a81bbbf8298c0fa03ea29cdc473d45769f953675",
+							},
+							Type:   checker.DependencyUseTypeGHAction,
+							Pinned: asBoolPointer(true),
+						},
+						{
+							Location: &checker.File{
+								Snippet: "other/checkout@a81bbbf8298c0fa03ea29cdc473d45769f953675",
+							},
+							Type:   checker.DependencyUseTypeGHAction,
+							Pinned: asBoolPointer(true),
+						},
+					},
+				},
+			},
+			outcomes: []finding.Outcome{
+				finding.OutcomePositive,
+				finding.OutcomePositive,
+			},
+		},
+		{
+			name: "GitHub Actions ecosystem with GitHub-owned and third-party pinned",
+			raw: &checker.RawResults{
+				PinningDependenciesResults: checker.PinningDependenciesData{
+					Dependencies: []checker.Dependency{
+						{
+							Location: &checker.File{
+								Snippet: "actions/checkout@v2",
+							},
+							Type:   checker.DependencyUseTypeGHAction,
+							Pinned: asBoolPointer(false),
+						},
+						{
+							Location: &checker.File{
+								Snippet: "other/checkout@v2",
+							},
+							Type:   checker.DependencyUseTypeGHAction,
+							Pinned: asBoolPointer(false),
+						},
+					},
+				},
+			},
+			outcomes: []finding.Outcome{
+				finding.OutcomeNegative,
+				finding.OutcomeNegative,
+			},
+		},
+		{
+			name: "GitHub Actions ecosystem with GitHub-owned pinned and third-party unpinned",
+			raw: &checker.RawResults{
+				PinningDependenciesResults: checker.PinningDependenciesData{
+					Dependencies: []checker.Dependency{
+						{
+							Location: &checker.File{
+								Snippet: "actions/checkout@a81bbbf8298c0fa03ea29cdc473d45769f953675",
+							},
+							Type:   checker.DependencyUseTypeGHAction,
+							Pinned: asBoolPointer(true),
+						},
+						{
+							Location: &checker.File{
+								Snippet: "other/checkout@v2",
+							},
+							Type:   checker.DependencyUseTypeGHAction,
+							Pinned: asBoolPointer(false),
+						},
+					},
+				},
+			},
+			outcomes: []finding.Outcome{
+				finding.OutcomePositive,
+				finding.OutcomeNegative,
+			},
+		},
+		{
+			name: "GitHub Actions ecosystem with GitHub-owned unpinned and third-party pinned",
+			raw: &checker.RawResults{
+				PinningDependenciesResults: checker.PinningDependenciesData{
+					Dependencies: []checker.Dependency{
+						{
+							Location: &checker.File{
+								Snippet: "actions/checkout@v2",
+							},
+							Type:   checker.DependencyUseTypeGHAction,
+							Pinned: asBoolPointer(false),
+						},
+						{
+							Location: &checker.File{
+								Snippet: "other/checkout@a81bbbf8298c0fa03ea29cdc473d45769f953675",
+							},
+							Type:   checker.DependencyUseTypeGHAction,
+							Pinned: asBoolPointer(true),
+						},
+					},
+				},
+			},
+			outcomes: []finding.Outcome{
+				finding.OutcomeNegative,
+				finding.OutcomePositive,
+			},
+		},
+		{
+			name: "Skipped objects and dependencies",
```

Comment:
what's skipped about this test?

Reply:
I don't know. I on purpose did not change test names to ensure compatibility. This is the test from here: 

https://github.com/ossf/scorecard/blob/b21a350f500a1ec4552ab40a1a9af236cfc06d97/checks/evaluation/pinned_dependencies_test.go#L801-L828